google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,502 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Wrappers for Document AI Document type."""
+
+import dataclasses
+import re
+import copy
+
+from typing import List, Tuple
+
+from google.cloud import documentai
+from google.cloud import storage
+
+from google.cloud.documentai_toolbox.wrappers.entity import Entity
+
+from google.cloud.documentai_toolbox.wrappers.document import Document
+
+_OP_TYPE = documentai.Document.Provenance.OperationType
+
+
+def _get_base_and_revision_bytes(
+    output_bucket: str, output_prefix: str
+) -> Tuple[str, list, list]:
+    r"""Returns document text,base docproto and list of revision shards
+
+    Args:
+        output_bucket (str):
+            Required. The name of the output_bucket.
+
+        output_prefix (str):
+            Required. The prefix of the folder where files are excluding the bucket name.
+
+    Returns:
+        Tuple[str, list, list]:
+            document text, base document.proto, revision shards.
+
+    """
+    text = ""
+    base_doc = []
+    revisions_doc = []
+
+    storage_client = storage.Client()
+
+    blob_list = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+    pb = documentai.Document.pb()
+    for blob in blob_list:
+        name = blob.name.split("/")[-1]
+        if blob.name.endswith(".dp.bp"):
+            blob_as_bytes = blob.download_as_bytes()
+            if re.search(r"^doc.dp.bp", name):
+                text = pb.FromString(blob_as_bytes).text
+            elif re.search(r"^pages_*.*", name):
+                base_doc.append(blob_as_bytes)
+            elif re.search(r"^rev_*.*", name):
+                revisions_doc.append(blob_as_bytes)
+
+    return text, base_doc, revisions_doc
+
+
+def _get_base_docproto(gcs_bucket_name, gcs_prefix) -> Tuple[list, list]:
+    r"""
+    Returns base document.proto and list of revision shards.
```

Comment:
Similarly, this docstring along with the return type annotation makes it sound like `base document.proto` is some kind of list.  I would discourage using the ambiguous term "document.proto", besides the confusion between the schema and individual messages, it also collides with the filename of https://github.com/googleapis/googleapis/blob/master/google/cloud/documentai/v1/document.proto, where the schema is defined.

Just to be sure, "document proto" isn't much better - are we referring to the schema or messages?  I would suggest just calling them:

- "documentai.Document messages" if we are talking about the proto-plus message instances
- "documentai.Document protobuf messages" if we are talking about the underlying protobuf messages (this should be rare)
- "bytes representing a documentai.Document message" if we are talking about the bytes underlying the above